segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -552,6 +571,144 @@ public void updateSegmentPolicy(UpdateSegmentPolicy updateSegmentPolicy) {
                 });
     }
 
+    @Override
+    public void createTableSegment(final CreateTableSegment createTableSegment) {
+        final String operation = "createTableSegment";
+
+        if (!verifyToken(createTableSegment.getSegment(), createTableSegment.getRequestId(), createTableSegment.getDelegationToken(), operation)) {
+            return;
+        }
+
+        log.info(createTableSegment.getRequestId(), "Creating table segment {}.", createTableSegment);
+        tableStore.createSegment(createTableSegment.getSegment(), TIMEOUT)
+                  .thenAccept(v -> connection.send(new SegmentCreated(createTableSegment.getRequestId(), createTableSegment.getSegment())))
+                  .exceptionally(e -> handleException(createTableSegment.getRequestId(), createTableSegment.getSegment(), operation, e));
+    }
+
+    @Override
+    public void deleteTableSegment(final DeleteTableSegment deleteTableSegment) {
+        String segment = deleteTableSegment.getSegment();
+        final String operation = "deleteTableSegment";
+
+        if (!verifyToken(segment, deleteTableSegment.getRequestId(), deleteTableSegment.getDelegationToken(), operation)) {
+            return;
+        }
+
+        log.info(deleteTableSegment.getRequestId(), "Deleting table segment {}.", deleteTableSegment);
+        tableStore.deleteSegment(segment, deleteTableSegment.isMustBeEmpty(), TIMEOUT)
+                  .thenRun(() -> connection.send(new SegmentDeleted(deleteTableSegment.getRequestId(), segment)))
+                  .exceptionally(e -> handleException(deleteTableSegment.getRequestId(), segment, operation, e));
+    }
+
+    @Override
+    public void mergeTableSegments(final MergeTableSegments mergeTableSegments) {
+        final String operation = "mergeTableSegments";
+
+        if (!verifyToken(mergeTableSegments.getSource(), mergeTableSegments.getRequestId(), mergeTableSegments.getDelegationToken(), operation)) {
+            return;
+        }
+
+        log.info(mergeTableSegments.getRequestId(), "Merging table segments {}.", mergeTableSegments);
+        tableStore.merge(mergeTableSegments.getTarget(), mergeTableSegments.getSource(), TIMEOUT)
```

Comment:
Do we also want to support conditional merge?

Reply:
tableStore.merge() is currently not implemented and will throw a `UnsupportedOperationException`